napari/components/_layer_slicer.py

```diff
@@ -43,9 +46,48 @@ def __init__(self):
         """
         self.events = EmitterGroup(source=self, ready=Event)
         self._executor: Executor = ThreadPoolExecutor(max_workers=1)
+        self._force_sync = False
         self._layers_to_task: Dict[Tuple[Layer], Future] = {}
         self._lock_layers_to_task = RLock()
 
+    @contextmanager
+    def force_sync(self):
+        """Context manager to temporarily force slicing to be synchronous.
+        This should only be called from the main thread.
+
+        >>> layer_slicer = _LayerSlicer()
+        >>> layer = Image(...)  # an async-ready layer
+        >>> with layer_slice.force_sync():
+        >>>     layer_slicer.slice_layers_async(layers=[layer], dims=Dims())
+        """
+        self._force_sync = True
+        yield None
+        self._force_sync = False
+
+    def wait_until_idle(self, timeout: Optional[float] = None) -> None:
+        """Wait for all slicing tasks to complete before returning.
+
+        Attributes
+        ----------
+        timeout: float or None
+            (Optional) time in seconds to wait before raising TimeoutError. If set as None,
+            there is no limit to the wait time. Defaults to None
+
+        Raises
+        ------
+        TimeoutError: when the timeout limit has been exceeded and the task is
+            not yet complete
+        """
+        futures = self._layers_to_task.values()
+        done, _ = wait(futures, timeout=timeout)
+
+        if done or not futures:
+            return
+        else:
+            raise TimeoutError(
+                f'Slicing tasks did not complete within timeout ({timeout}s).'
+            )
```

Comment:
You don't need the early return case here since this is the last code of the function. I'd also suggest expanding the condition and name of the variable to make it clear this is a sequence without needing to go to the Python docs for `wait`.

```suggestion
        _, not_done_futures = wait(futures, timeout=timeout)
        
        if len(not_done_futures) > 0:
            raise TimeoutError(
                f'Slicing tasks did not complete within timeout ({timeout}s).'
            )
```